build/debian-iptables/iptables-wrapper.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# Detect whether the base system is using iptables-legacy or
+# iptables-nft. This assumes that some non-containerized process (eg
+# kubelet) has already created some iptables rules.
+num_legacy_lines=$( (iptables-legacy-save || true; ip6tables-legacy-save || true) 2>/dev/null | wc -l)
+num_nft_lines=$( (iptables-nft-save || true; ip6tables-nft-save || true) 2>/dev/null | wc -l)
+if [ "${num_legacy_lines}" -gt "${num_nft_lines}" ]; then
+    mode=legacy
+else
+    mode=nft
+fi
+
+update-alternatives --set iptables "/usr/sbin/iptables-${mode}" > /dev/null
```

Comment:
Is there a reason to update-alternatives vs just execing the correct mode binary directly?

Reply:
It moves the wrapper script out of the way so that all future iptables calls will go directly to the correct binary. (We don't want kube-proxy to have to run "`iptables-save`" an extra time to figure out the correct mode every time it pushes an update.)